lib/typecheck.ts

```diff
@@ -679,6 +679,61 @@ export default class TypeChecker {
         return Promise.resolve();
     }
 
+    private async _typeCheckJoin(ast : Ast.JoinExpression, scope : Scope) {
+        assert(ast.lhs.schema && ast.rhs.schema);
+        const name = `join(${ast.lhs.schema.name},${ast.rhs.schema.name})`;
+        if (!ast.lhs.schema.is_list || !ast.rhs.schema.is_list)
+            throw new TypeError('Both sides of a join must be a list query');
+        const classDef = null; // FIXME: what should this be for a join expression? 
```

Comment:
It definitely should be null (this will cause issues in the dialogue code but we'll handle those)

Reply:
comment removed in 2f7b9f5f